src/controllers/services.js

```diff
@@ -0,0 +1,101 @@
+import { fetchCollections } from '../services/collections.js';
+import { fetchServices } from '../services/services.js';
+
+export const getServices = async (req, res) => {
+  const { name, termsTypes } = req.query;
+
+  const collections = await fetchCollections().catch(error => {
+    res.status(error.response?.status || 500).json({ error: `${error.message} (${error.code})` });
+  });
+
+  if (!collections) {
+    return;
+  }
+
+  const results = [];
+  const failures = [];
+
+  await Promise.all(collections.map(async ({ id: collectionId, endpoint }) => {
+    const services = await fetchServices(endpoint).catch(error => {
+      failures.push({ collection: collectionId, message: error.toString() });
+    });
+
+    if (!services) {
+      return;
+    }
+
+    for (const service of services) {
+      if (!service.terms || (name && service.name !== name)) {
+        continue;
+      }
+
+      const serviceTermsTypes = service.terms.map(terms => terms.type);
+      const filteredServiceTermsTypes = termsTypes ? serviceTermsTypes.filter(serviceTermsType => termsTypes.includes(serviceTermsType)) : serviceTermsTypes;
+
+      if (!filteredServiceTermsTypes.length) {
+        continue;
+      }
```

Comment:
🧠 Once more, with a functional approach, keeping the same variable all along:
```suggestion
      if (termsTypes) {
        services = services.filter(service => service.terms.find(terms => termsTypes.includes(terms.type)));
      }
```